src/views/Nft/market/hooks/useCollectionNfts.tsx

```diff
@@ -0,0 +1,161 @@
+  fetchNftsFiltered,
+  getMarketDataForTokenIds,
+  getNftApi,
+  getNftsFromCollectionApi,
+  getNftsMarketData,
+} from 'state/nftMarket/helpers'
+// eslint-disable-next-line camelcase
+import useSWRInfinite from 'swr/infinite'
+import isEmpty from 'lodash/isEmpty'
+import uniqBy from 'lodash/uniqBy'
+import { REQUEST_SIZE } from '../Collection/config'
+
+interface ItemListingSettings {
+  field: string
+  direction: 'asc' | 'desc'
+  showOnlyNftsOnSale: boolean
+  nftFilters: Record<string, NftAttribute>
+}
+
+const fetchTokenIdsFromFilter = async (address: string, settings: ItemListingSettings) => {
+  const filterObject: Record<string, NftAttribute> = settings.nftFilters
+  const attrParams = Object.values(filterObject).reduce(
+    (accum, attr) => ({
+      ...accum,
+      [attr.traitType]: attr.value,
+    }),
+    {},
+  )
+  const attrFilters = !isEmpty(attrParams) ? await fetchNftsFiltered(address, attrParams) : null
+  return attrFilters ? Object.values(attrFilters.data).map((apiToken) => apiToken.tokenId) : null
+}
+
+const fetchMarketDataNfts = async (address: string, settings: ItemListingSettings, page: number) => {
+  const tokenIdsFromFilter = await fetchTokenIdsFromFilter(address, settings)
+  const whereClause = tokenIdsFromFilter
+    ? {
+        collection: address.toLowerCase(),
+        isTradable: true,
+        tokenId_in: tokenIdsFromFilter,
+      }
+    : { collection: address.toLowerCase(), isTradable: true }
+  const subgraphRes = await getNftsMarketData(
+    whereClause,
+    REQUEST_SIZE,
+    settings.field,
+    settings.direction,
+    page * REQUEST_SIZE,
+  )
+  const apiRequestPromises = subgraphRes.map((marketNft) => getNftApi(address, marketNft.tokenId))
+  const apiResponses = await Promise.all(apiRequestPromises)
+  const newNfts = apiResponses.reduce((acc, apiNft) => {
+    if (apiNft) {
+      acc.push({
+        ...apiNft,
+        collectionAddress: address,
+        collectionName: apiNft.collection.name,
+        marketData: subgraphRes.find((marketNft) => marketNft.tokenId === apiNft.tokenId),
+      })
+    }
+    return acc
+  }, [])
+  return newNfts
+}
+
+const fetchAllNfts = async (address: string, settings: ItemListingSettings, page: number) => {
+  const tokenIdsFromFilter = await fetchTokenIdsFromFilter(address, settings)
+
+  let collectionNftsResponse: ApiResponseCollectionTokens = null
+  let tokenIds = null
+
+  if (tokenIdsFromFilter) {
+    tokenIds = tokenIdsFromFilter
+  } else {
+    collectionNftsResponse = await getNftsFromCollectionApi(address, REQUEST_SIZE, page)
+    if (collectionNftsResponse?.data) {
+      tokenIds = Object.values(collectionNftsResponse.data).map((nft) => nft.tokenId)
+    }
+  }
+
+  if (tokenIds) {
+    const nftsMarket = await getMarketDataForTokenIds(address, tokenIds)
+
+    const responsesPromises = tokenIds.map(async (id) => {
+      const apiMetadata = collectionNftsResponse ? collectionNftsResponse.data[id] : await getNftApi(address, id)
```

Comment:
`apiMetadata` type should be explicit because we don't know what results are returned. It might lead to TypeError

Reply:
Done